arch: arm64: Remove useless asm code

carlocaione opened the pull request:
The content of the SCR_EL3 register is overwritten by a later instruction. Also no need to route SError, IRQs and FIQs to EL3.

carlocaione committed:
arch: arm64: Remove useless assembly code

The content of the SCR_EL3 register is overwritten by a later
instruction. Also no need to route SError, IRQs and FIQs to EL3.

Signed-off-by: Carlo Caione <ccaione@baylibre.com>

sandeepbrcm reviewed (commented):
I think we missed this in previous review.
Till the time control has not switched to lower EL we should be able to catch any abort arising out of` EL3` code.
Instead of removing this we may have to put back the 'z_fatal_error()' handling `esr_eln` dump to consider `EL3` as well.   `IRQ`  or `FIQ`  are not possible to come at this point. But still as a convention
we can give a valid configuration. ie: if interrupt comes ( I would consider that as hardware glitch) then handle ( which is b .) at EL3 vector because there is no other option.

carlocaione commented:
> I think we missed this in previous review.
> Till the time control has not switched to lower EL we should be able to catch any abort arising out of` EL3` code.
> Instead of removing this we may have to put back the 'z_fatal_error()' handling `esr_eln` dump to consider `EL3` as well. `IRQ` or `FIQ` are not possible to come at this point. But still as a convention

This was not overlooked but the chances that an abort occurs during the reset routines are really low (probably only in case of a faulty hardware). I can adjust the PR as you suggested but this is probably gonna be over-engineered :)

> we can give a valid configuration. ie: if interrupt comes ( I would consider that as hardware glitch) then handle ( which is b .) at EL3 vector because there is no other option.

Interrupts should be masked by the `DAIF` register at this stage.

carlocaione committed:
arch: aarch64: Catch early errors in EL3 and EL1

Setup the stack as early as possible to catch any possible errors in the
reset routine and handle also EL3 fatal errors.

Signed-off-by: Carlo Caione <ccaione@baylibre.com>

sandeepbrcm commented:
> Interrupts should be masked by the `DAIF` register at this stage.

Yeah, right. we need not trap them. Pls ignore `I/F`.

> chances that an abort occurs during the reset routines are really low

But I think its worth to retain the capability especially considering the platform hook can be extended to add any code at EL3.

sandeepbrcm commented on arch/arm/core/aarch64/reset.S:
```diff
@@ -62,6 +62,11 @@ SECTION_SUBSEC_FUNC(TEXT,_reset_section,__start)
 	msr	vbar_el3, x19
 	isb
 
+	/* Setup a stack for EL3 (SP_ELx) */
```
+1

sandeepbrcm commented on arch/arm/core/aarch64/reset.S:
```diff
@@ -121,6 +126,12 @@ SECTION_SUBSEC_FUNC(TEXT,_reset_section,__start)
 	msr	vbar_el1, x19
+	isb
+
+	/* Setup the stack (SP_ELx) */
```
+1

carlocaione commented:
`buildkite` failure unrelated (I hope). Force pushing again to re-trigger the CI.
